src/cache/cacheUpdator.js

```diff
@@ -0,0 +1,45 @@
+const axios = require("axios");
+const { newsURL } = require("../config");
+const users = require("../data/users.json");
+const userNewsPreferences = require("../data/userNewsPreferences.json");
+const redisClient = require("../cache/redis");
+
+const updateNewsCache = () => {
+  const parsedUserPreferences = JSON.parse(JSON.stringify(userNewsPreferences));
+  setInterval(() => {
+    users.forEach(async (user) => {
+      await redisClient
+        .ttl(`news-${user.id}`)
+        .then((expiredTime) => {
+          const userPreference = parsedUserPreferences[user.id];
```

Comment:
We can check this before making a call to the Redis cache for checking TTL, as you would be doing an extra call over your network to fetch this in case of a real-world system.